lib/plugins/helper/toc.ts

```diff
@@ -102,4 +104,26 @@ function tocHelper(str: string, options: Options = {}) {
   return result;
 }
 
+function getAndTruncateTocObj(str: string, options: {min_depth: number, max_depth: number}, max_items: number) {
+  let data = tocObj(str, { min_depth: options.min_depth, max_depth: options.max_depth });
+
+  if (data.length === 0) {
+    return data;
+  }
+  if (max_items < 1) {
+    return data;
+  }
+
+  const min = Math.min(...data.map(item => item.level));
+  const max = Math.max(...data.map(item => item.level));
```

Comment:
`...data.map(item => item.level)` They look the same, 
I guess it doesn't need to be executed twice

Reply:
fixed: f8d04a6